crates/object_detection/src/pose_interpretation.rs

```diff
@@ -0,0 +1,256 @@
+                context
+                    .hardware_interface
+                    .write_to_network(OutgoingMessage::Spl(HulkMessage {
+                        player_number: *context.player_number,
+                        fallen: matches!(context.fall_state, FallState::Fallen { .. }),
+                        pose: ground_to_field.as_pose(),
+                        over_arms_pose_detected: true,
+                        ball_position: None,
+                        time_to_reach_kick_position: Some(*context.time_to_reach_kick_position),
+                    }))?;
+            }
+        };
+
+        context.detected_pose_types.fill_if_subscribed(|| {
+            Self::get_all_pose_types(
+                human_poses,
+                context.camera_matrices.top.clone(),
+                context.ground_to_field,
+                *context.foot_z_offset,
+                *context.keypoint_confidence_threshold,
+                *context.shoulder_angle_threshold,
+            )
+        });
+
+        Ok(MainOutputs {
+            detected_referee_pose_type: pose_type.into(),
+        })
+    }
+
+    pub fn get_all_pose_types(
+        poses: &[HumanPose],
+        camera_matrix_top: CameraMatrix,
+        ground_to_field: Option<&Isometry2<Ground, Field>>,
+        foot_z_offset: f32,
+        keypoint_confidence_threshold: f32,
+        shoulder_angle_threshold: f32,
+    ) -> Vec<(PoseType, Point2<Field>)> {
+        poses
+            .iter()
+            .filter_map(|pose| {
+                let ground_to_field = ground_to_field?;
+                let left_foot_ground_position = camera_matrix_top
+                    .pixel_to_ground_with_z(pose.keypoints.left_foot.point, foot_z_offset)
+                    .ok();
+                let right_foot_ground_position = camera_matrix_top
+                    .pixel_to_ground_with_z(pose.keypoints.right_foot.point, foot_z_offset)
+                    .ok();
+                let (left_foot_ground_position, right_foot_ground_position) =
+                    left_foot_ground_position.zip(right_foot_ground_position)?;
+                let interpreted_pose = Self::interpret_pose(
+                    &Some(*pose),
+                    keypoint_confidence_threshold,
+                    shoulder_angle_threshold,
+                );
+                Some((
+                    interpreted_pose,
+                    center(
+                        ground_to_field * left_foot_ground_position,
+                        ground_to_field * right_foot_ground_position,
+                    ),
+                ))
+            })
+            .collect()
+    }
+
+    pub fn get_referee_pose(
+        poses: &[HumanPose],
+        camera_matrix_top: CameraMatrix,
+        distance_to_referee_position_threshold: f32,
+        expected_referee_position: Point2<Ground>,
+        foot_z_offset: f32,
+    ) -> Option<HumanPose> {
+        let pose_type_tuple = poses
```

Comment:
Do not repeat yourself: do not name the types of the variables in the variable name

Reply:
Done